include/alpaka/warp/Traits.hpp

```diff
@@ -0,0 +1,193 @@
+/* Copyright 2020 Sergei Bastrakov
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+#include <alpaka/core/Concepts.hpp>
+
+#include <cstdint>
+#include <type_traits>
+
+namespace alpaka
+{
+    //-----------------------------------------------------------------------------
+    //! The thread warp specifics
+    namespace warp
+    {
+        struct ConceptWarp{};
+
+        //-----------------------------------------------------------------------------
+        //! The warp traits.
+        namespace traits
+        {
+            //#############################################################################
+            //! The warp size trait.
+            template<
+                typename TWarp,
+                typename TSfinae = void>
+            struct GetSize;
+
+            //#############################################################################
+            //! The all warp vote trait.
+            template<
+                typename TWarp,
+                typename TSfinae = void>
+            struct All;
+
+            //#############################################################################
+            //! The any warp vote trait.
+            template<
+                typename TWarp,
+                typename TSfinae = void>
+            struct Any;
+
+            //#############################################################################
+            //! The ballot warp vote trait.
+            template<
+                typename TWarp,
+                typename TSfinae = void>
+            struct Ballot;
+
+            //#############################################################################
+            //! The active mask trait.
+            template<
+                typename TWarp,
+                typename TSfinae = void>
+            struct Activemask;
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Returns warp size.
+        //!
+        //! \tparam TWarp The warp implementation type.
+        //! \param warp The warp implementation.
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<
+            typename TWarp>
+        ALPAKA_FN_ACC auto getSize(
+            TWarp const & warp)
+        -> std::int32_t
+        {
+            using ImplementationBase = concepts::ImplementationBase<
+                ConceptWarp,
+                TWarp>;
+            return traits::GetSize<
+                ImplementationBase>
+            ::getSize(
+                warp);
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Evaluates predicate for all active threads of the warp and returns
```

Comment:
We need to check if this is correct doxygen. IMO `//!` is for the brief 

https://www.star.bnl.gov/public/comp/sofi/doxygen/docblocks.html

Reply:
I always use `//! ` and Doxygen does not differentiate at all.

Reply:
I think I copied this style from some Alpaka file. But do not remember, perhaps the original was indeed a brief, but my comments should certainly not be one.

Reply:
Okay, after reading the document linked I think this style is actually fine, and is not a brief. (Covered by point 3 of the doc).